lib/parser/gpx.rb

```diff
@@ -42,6 +46,40 @@ def parse # rubocop:disable Metrics/AbcSize
 
     private
 
+    def enhance_trace(points)
+      prev_point = points.first
+      points[1..].each_with_object([]) do |cur_point, et|
+        duration = cur_point[:time].to_f - prev_point[:time].to_f
+        
+        prev = Geokit::LatLng.new(prev_point[:lat].to_f, prev_point[:lon].to_f)
+        cur  = Geokit::LatLng.new(cur_point[:lat].to_f, cur_point[:lon].to_f)
+        distance = prev.distance_to(cur)
+        
+        speed = distance / duration / 1000 * 3600
+        
+        et << cur_point.merge(distance: distance, duration: duration, speed: speed)
+        prev_point = cur_point
+      end
+    end
+
+    def set_thresholds(points)
+      # calculate average speed
+      # and assume to slow is 1/10th of it
+      avg_speed = points.sum { |p| p[:speed] } / points.size
+      @to_slow = avg_speed / 10
```

Comment:
It would be interesting to check if that would still work with ski touring. there you are veeery slow when going up, but pretty fast going down

Reply:
this one for example: https://www.runtastic.com/de/benutzer/73412fb6-067f-ae70-caba-25969cfd75d2/sportaktivitaeten/3e8d10d8-1a75-4ae4-8e0b-fa8d02663677 it contains a long pause between 3:52 and 5:06 and two others between 3:00 - 3:25 and 2:08 - 2:21

Reply:
Can you try it ?

Reply:
I guess there will be some cases where the pause-detection does not work properly .. would be awesome to get examples and then enhance. :)